compute_endpoint/globus_compute_endpoint/endpoint/rabbit_mq/result_publisher.py

```diff
@@ -0,0 +1,358 @@
+            mq_chan.connection.ioloop.call_later(1, self._open_channel)
+
+        else:
+            log.error(
+                f"{self} Unable to sustain channel after {len(self._channel_closes)}"
+                f" attempts in {self.channel_close_window_limit} seconds. ({exc})"
+            )
+            self._stop_event.set()
+
+    def _on_exchange_verified(self, _frame: Method[Exchange.DeclareOk]):
+        assert self._mq_chan is not None
+        self._mq_chan.queue_declare(
+            passive=True,
+            queue=self.queue_info["queue"],
+            callback=self._on_queue_verified,
+        )
+
+    def _on_queue_verified(self, _frame: Method[pikaQueue.DeclareOk]):
+        assert self._mq_chan is not None
+        self._mq_chan.confirm_delivery(self._on_delivery)
+        self.status = RabbitPublisherStatus.connected
+        self._connected_at = time.time()
+
+    def _on_delivery(self, frame: Method):
+        is_ack = frame.method.INDEX == Basic.Ack.INDEX
+        is_nack = frame.method.INDEX == Basic.Nack.INDEX
+        is_multiple = frame.method.multiple
+        frame_dtag = frame.method.delivery_tag
+
+        f = self._published.pop(frame_dtag)
+        if is_ack:
+            f.set_result(None)
+
+            if is_multiple:
+                for dtag in list(self._published.keys()):
+                    if dtag < frame_dtag:
+                        f = self._published.pop(dtag)
+                        f.set_result(None)
+        elif is_nack:
+            exc = Exception(f"Failed to publish delivery_tag {frame_dtag}")
+            f.set_exception(exc)
+            if is_multiple:
+                for dtag in list(self._published.keys()):
+                    if dtag < frame_dtag:
+                        f = self._published.pop(dtag)
+                        exc = Exception(f"Failed to publish delivery_tag {dtag}")
+                        f.set_exception(exc)
+
+    def _event_watcher(self):
+        if self._stop_event.is_set():
+            log.info("%r Shutting down due to stop event set.")
+            self._stop_ioloop()
+            return
+
+        if self._connection_tries and self._connected_at:
+            # we're connected ...
+            if time.time() - self._connected_at > 60:
+                # ... and connection stable for 60s; good to reset connection tries
+                self._connection_tries = 0
+                log.debug("%r Connection deemed stable; resetting connection tally")
+
+        try:
+            if self.status != RabbitPublisherStatus.connected:
+                log.debug("%r not yet connected; holding enqueued messages")
+                raise queue.Empty()
+
+            while True:
+                f, msg_bytes = self._outstanding.get(block=False)
+                try:
+                    self._mq_chan.basic_publish(body=msg_bytes, **self._publish_kwargs)
+                    self._delivery_tag_index += 1
+                    self._published[self._delivery_tag_index] = f
```

Comment:
I'd suggest renaming this variable from `self._published` to something like `self._awaiting_confirmation`?

Reply:
Done.  Nice thought process.  Clarity wins!